zerver/tests/test_event_queue.py

```diff
@@ -760,6 +760,33 @@ def test_stream_email_and_push_notify_unmuted_topic_and_unmuted_stream(
                 already_notified={"email_notified": False, "push_notified": False},
             )
 
+    def test_followed_topic_push_notify_global_setting(self) -> None:
+        do_change_user_setting(
+            self.user_profile, "enable_followed_topic_push_notifications", False, acting_user=None
+        )
+        do_set_user_topic_visibility_policy(
```

Comment:
Similar to this, I think we should also add a test in the previous commit to verify that the email setting when disabled does not send notifications.

Reply:
Organized the new tests in this file in a better way. (While working on the wildcard mention part, I observed the better way in which these new tests can be organized).

[Newly added tests](https://github.com/zulip/zulip/pull/25640/commits/c2e10004496fbf0dcccf210f4c1ae159f9bfe56e#diff-b7ecab40a23cd87e0acf58e7c8220330787bb4f7bb59e8ba601ae648ffdc968aR763)